elide-core/src/main/java/com/yahoo/elide/graphql/PersistentResourceFetcher.java

```diff
@@ -263,12 +267,382 @@ private Object fetchRelationship(Environment context) {
+                        toVisit.add(new Triplet((List<Map<String, Object>>) nestedRelation.getValue(),
+                                relationshipResource.iterator().next(), nestedRelation.getKey()));
+                    }
+                }
+
+                /* update the relationship between parent and newly created/updated nested resources */
+                newParent.updateRelation(relationship.getKey(), relationshipResource);
+                returnOps.addAll(relationshipResource);
+            }
+        }
+        return returnOps;
+    }
+
+    /**
+     * updates or creates existing/new entities
+     * @param input data input
+     * @param id id
+     * @param requestScope request scope
+     * @param outputType graphql output type
+     * @param fieldName field name
+     * @param parent parent resource
+     * @return {@link PersistentResource} object
+     */
+    private PersistentResource updateOrCreateAttributes(Map<String, Object> input, Optional<String> id,
+                                              RequestScope requestScope, Class<?> loadClass,
+                                              GraphQLType outputType, String fieldName,
+                                              PersistentResource parent) {
+        if(!id.isPresent()) { /* no id(s) provided, must create new */
+            return createObject(input, fieldName, requestScope, Optional.empty(), outputType, parent);
+        }
+
+        Set<PersistentResource> fetchEntries;
+        try {
+            if(parent == null) {
+                fetchEntries = fetchObject(requestScope, loadClass,
+                        Optional.of(Arrays.asList(id.orElse(null))), Optional.empty(),
+                        Optional.empty(), Optional.empty(), Optional.empty());
+            } else {
+                fetchEntries = (Set<PersistentResource>) fetchRelationship(parent, fieldName);
+            }
+        } catch (BadRequestException e) {
+            throw e;
+        }
+
+        if(fetchEntries.isEmpty() || fetchEntries == null) { /* empty set returned, must create new */
+            return createObject(input, fieldName, requestScope, Optional.empty(), outputType, parent);
+        } else { /* must update object */
+            return updateObject(fetchEntries, input, fieldName, requestScope, parent);
+        }
+    }
+
+    /**
+     * Strips out the attributes from the {@code element} singleton list and returns just the relationships, if present.
+     * @param element data input
+     * @param entityClass entity class
+     * @param dictionary entity dictionary
+     * @return relationship map
+     */
+    private Map<String, Object> stripAttributes(List<Map<String, Object>> element, Class<?> entityClass,
+                                                EntityDictionary dictionary) {
+        Map<String, Object> relationshipsOnly = new HashMap<>();
+        Map<String, Object> entry = element.get(0); /* 'element' will always be singleton */
```

Comment:
At this level of function (low level helper), it doesn't have the context to know that the list will only have a single element.  That decision should happen somewhere else.  I would change this function to not take a list of maps, but rather just a single map.